### PR TITLE
Bump main quarkus 2.14

### PR DESCRIPTION
**Pull request by mbiarnes**
Please review and merge
- https://github.com/kiegroup/kogito-images/pull/1357
- https://github.com/kiegroup/kogito-docs/pull/244
- https://github.com/kiegroup/drools/pull/4776
- https://github.com/kiegroup/kogito-runtimes/pull/2583
- https://github.com/kiegroup/kogito-examples/pull/1452

### CODE REVIEW

**Comment by kie-ci1** [comment]
**PR job** `#343` was: **FAILURE**
Possible explanation: Pipeline failure or project build failure

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/343/display/redirect
See console log:
<details>
<summary><b>Console Logs</b></summary>
[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] }<br/>[Pipeline] // withEnv<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Prepare for tests)<br/>Stage "Prepare for tests" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Build & Test Images)<br/>Stage "Build & Test Images" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Declarative: Post Actions)<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>+ rm -rf /home/jenkins/workspace/KIE/kogito/main/pullrequest/kogito-images/workspaces<br/>[Pipeline] sh<br/>++ docker ps -a -q<br/>+ docker rm -f 6a377ce5182a<br/>6a377ce5182a<br/>[Pipeline] sh<br/>++ docker images -q<br/>+ docker rmi -f 163907a855f2<br/>Untagged: sonatype/nexus3:latest<br/>Untagged: sonatype/nexus3@sha256:7e7abd3418d507d5263460eda83e239aff758cd362f8add54d9c9846cada2533<br/>Deleted: sha256:163907a855f233f59aa8fea98a5bf6e9e80924fec48c60d63b6597ece596e655<br/>Deleted: sha256:0436e60d5233909d691b289d9348f7a419142815f1f2ac914bcb49cdce558a1a<br/>Deleted: sha256:285710bf5f2846cba7e0dfc49d971e26ca3c6709f4d008938cdd181cf1765020<br/>Deleted: sha256:915a22a3d7d4148368570458bae2f34312f84ba255e4e7b638318428eebd707b<br/>Deleted: sha256:00623169088beff3a4e49f2a909e9caa1b0d7385ee7a2c82151b216af434f6b4<br/>Deleted: sha256:678b51c83167cf0f29ec13c18c2eaf0c73ab7c9ddb8003fae201f8b5614b8d61<br/>Deleted: sha256:ef1e25cd89b85afcd114a9c199883e17d7155a2038860ebcc9408bb4a74e3db0<br/>Deleted: sha256:e3b9b66ea6b08feea8b7aba285deed9964ebfc96c7b3286c8ef5544fe87fe87c<br/>Deleted: sha256:a138a8676cac6a1adffd6f1fe5d9ee21730864f9360472df3458533efc2033f7<br/>[Pipeline] sh<br/>+ rm -rf /home/jenkins/.cekit/cache<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>+ wget --no-check-certificate -qO - https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/343/api/json<br/>[Pipeline] readJSON<br/>[Pipeline] sh<br/>
</details>

**Comment by mbiarnes** [comment]
- created locally a new branch `bump_main_quarkus_2.14` based on recent main
- closed the `old` PR #1357
- removed branch
- cherry-picked all PRs of Tristan to the local branch
-  pushed and raised thsi PR

**Review by evacchi — APPROVED** [review]

File: modules/kogito-s2i-core/module.yaml [hunk 1]
```diff
@@ -42,10 +42,6 @@ envs:
     description: ^ Defines the maven artifact for Quarkus Maven plugin, defaults to io.quarkus:quarkus-maven-plugin. If no version specified it will be detected automatically based on the current Kogito version.
     example: "io.quarkus:custom-plugin:custom.version"
 
-  - name: QUARKUS_VERSION
```
Comment (spolti) [inline modules/kogito-s2i-core/module.yaml:45]: why it has been removed from here?

File: modules/kogito-swf-builder/module.yaml [hunk 2]
```diff
@@ -4,28 +4,31 @@ version: "2.0.0-snapshot"
 description: "Kogito Serverless Workflow builder with required extensions"
 
 envs:
+  - name: QUARKUS_EXTENSIONS
+    example: 'quarkus-kubernetes,kogito-quarkus-serverless-workflow,kogito-addons-quarkus-knative-eventing'
+    description: To add extension to your application
   - name: PROJECT_GROUP_ID
     value: "org.acme"
-    description: Please do not change
+    description: Please do not change. To change only if you plan to use the ${KOGITO_HOME}/launch/create-app.sh
```
Comment (spolti) [inline modules/kogito-swf-builder/module.yaml:12]:
```suggestion
    description: Change only if you plan to use the ${KOGITO_HOME}/launch/create-app.sh script
```
is there any example on how to use it?

File: modules/kogito-swf-builder/module.yaml [hunk 3]
```diff
@@ -4,28 +4,31 @@ version: "2.0.0-snapshot"
...
   - name: PROJECT_ARTIFACT_ID
     value: "serverless-workflow-project"
-    description: Please do not change
+    description: Please do not change. To change only if you plan to use the ${KOGITO_HOME}/launch/create-app.sh
```
Comment (spolti) [inline modules/kogito-swf-builder/module.yaml:15]:
```suggestion
    description: Change only if you plan to use the ${KOGITO_HOME}/launch/create-app.sh
```

**Comment by radtriste** [comment]
This will be dealt in https://github.com/kiegroup/kogito-images/pull/1361

**Pull request closed without merging**